.github/workflows/commitlint.yml

```diff
@@ -0,0 +1,17 @@
+name: Validate Commit Messages
+on:
+  [pull_request]
```

Comment:
+1 to running it as early as possible, not only for things targeting master. 
This may occasionally be annoying when doing messy experiments, but it's totally worth it.

Reply:
lol, I just reverted this.  I'm not sure it is worth it, honestly.  but I'll put it back.  

note that it does not run on `push` because, well, if you've pushed to `main` it's too late anyway.